substrate/client/network/sync/src/strategy/chain_sync.rs

```diff
@@ -337,6 +337,9 @@ pub struct ChainSync<B: BlockT, Client> {
 	import_existing: bool,
 	/// Block downloader
 	block_downloader: Arc<dyn BlockDownloader<B>>,
+	/// Is block pruning enabled? This indicates that the user is not
+	/// interested in historical blocks, so gap sync will be skipped.
+	block_pruning_enabled: bool,
```

Comment:
nit: adding this seems like duplication considering there is also an `Option<GapSync<B>>` field on this struct. Maybe some refactoring can make the `gap_sync` field directly configurable, while not needing to store the `block_pruning_enabled`. I haven't investigated what it involves though, might be a little cumbersome.